rust/kernel/types.rs

```diff
@@ -247,3 +247,53 @@ impl<T: FnOnce()> Drop for ScopeGuard<T> {
         }
     }
 }
+
+/// Trait for wrappers which are holding a direct pointer that they are getting from a C side function.
```

Comment:
Is this a new `PointerWrapper`?

Reply:
Yes, types implementing this trait only have a pointer, and are not Owned like the PointerWrapper.